libs/vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -567,13 +587,17 @@ async def _astream(
         if not self._is_gemini_model:
             raise NotImplementedError()
         params = self._prepare_params(stop=stop, stream=True, **kwargs)
-        history_gemini = _parse_chat_history_gemini(
+        chat_history_result = _parse_chat_history_gemini(
             messages,
             project=self.project,
             convert_system_message_to_human=self.convert_system_message_to_human,
         )
+        history_gemini = chat_history_result["history"]
+        system_instructions = chat_history_result["system_instructions"]
         message = history_gemini.pop()
         with telemetry.tool_context_manager(self._user_agent):
+            if system_instructions is not None:
```

Comment:
I don't think we need this if (if it's None, let's reset it to None then otherwise we might cache system instructions from the previous call)